src/troute-network/troute/AbstractNetwork.py

```diff
@@ -654,3 +658,201 @@ def _try_parsing_date(text):
             "channel initial states complete in %s seconds."\
             % (time.time() - start_time)
         )
+
+    def build_forcing_sets(self,):
```

Comment:
Should probably look at factoring out the common code and the network specific code in this function.